### PR TITLE
Limit switch workaround

### LINKED ISSUE #48: Temporarily work around limit switches
It looks like the bot we need to test on on Monday may not have limit switches attached. We need to implement **commands** (We should NOT have to change subsystems or existing commands! We should only implement NEW commands to do this!) that will allow us to test the basic functionality of the bot without having limit switches attached.

Keep in mind that instantiating objects for limit switches that aren't connected won't cause any issues, since the RIO has no way to know that the limit switch isn't actually there.

### PR DESCRIPTION
**Pull request by Jack-Haefele**
closes #48

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -124,6 +128,8 @@ private void configureBindings() {
     driverController.x().onTrue(enterXMode);
     driverController.a().onTrue(runIntake);
```
Comment (Berdenson) [inline src/main/java/frc/robot/RobotContainer.java:129]: remove the runIntake button mapping. We don't want A to run 2 very similar commands at the same time.

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -124,6 +128,8 @@ private void configureBindings() {
...
     driverController.y().onTrue(fireNote);
+    driverController.b().onTrue(deindexNoteNoLimitSwitch);
+    driverController.a().onTrue(RunIntakeNoLimitSwitch);
   }
```
Comment (Berdenson) [inline src/main/java/frc/robot/RobotContainer.java:133]: We also need an additional button mapping that runs deployMailbox.

File: src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    intake.runIntake();
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    /* If the mailbox is fully raised, run the intake. */
```
Comment (Berdenson) [inline src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java:41]: Remove unused comments

File: src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.mailbox.Mailbox;
+
+/** Outputs the note from the index belts into the mailbox belts. */
+public class DeindexNoteNoLimitSwitch extends Command {
+
+  private Intake intake;
+  private Mailbox mailbox;
```
Comment (Berdenson) [inline src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java:23]: Remove mailbox as we don't use it in this class

**Review by Berdenson — CHANGES_REQUESTED** [review]
Redo javadoc on copied files

File: src/main/java/frc/robot/commands/RunIntakeNoLimitSwitch.java
Comment (willitcode) [inline src/main/java/frc/robot/commands/RunIntakeNoLimitSwitch.java:1]: Don't make a new command for this. The intake code shouldn't need to be changed at all. Think about it: the only thing not having a limit switch on the intake will affect is that the command won't automatically end—so the driver will need to let go of the button quickly.

**Review by QuackitsQuinn — COMMENTED** [review]
It looks fine to me

File: src/main/java/frc/robot/commands/RunIntakeNoLimitSwitch.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * The Third Law: A robot must protect its own existence as long as such protection does not conflict with the First or Second Law.
+ */
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.Intake;
+
+/** Command that runs the intake on the robot. */
```
Comment (Berdenson) [inline src/main/java/frc/robot/commands/RunIntakeNoLimitSwitch.java:18]: Redo Javadoc for all copied classes.

**Review by Berdenson — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java
Comment (willitcode) [inline src/main/java/frc/robot/commands/mailbox/DeindexNoteNoLimitSwitch.java:1]: Delete this whole file. Instead, replace DeindexNoteNoLimitSwitch in FireNoteRoutineNoLimitSwitch with a simple RunIntake command—they'll do the same thing in this context.

File: src/main/java/frc/robot/commands/mailbox/FireNoteRoutineNoLimitSwitch.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+
+/**
+ * Scores the note into the amp. Raises the mailbox, runs the mailbox belts, and then runs the index
+ * belts to send the note into the belts.
+ */
+public class FireNoteRoutineNoLimitSwitch extends ParallelCommandGroup {
+  /** Creates a new FireNote. */
+  public FireNoteRoutineNoLimitSwitch() {
+    addCommands(new DeployMailbox(), new DeindexNoteNoLimitSwitch());
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/mailbox/FireNoteRoutineNoLimitSwitch.java:23]: Just use a RunIntake for the second command.

**Review by willitcode — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java
Comment (willitcode) [inline src/main/java/frc/robot/RobotContainer.java:1]: Bind the FireNoteRoutineNoteRoutineNoLimitSwitch command to a button

Comment (willitcode) [reply src/main/java/frc/robot/RobotContainer.java:1]: Leave the regular FireNoteRoutine button alone tho—we want buttons for both

**Review by willitcode — CHANGES_REQUESTED** [review]

**Review by Berdenson — APPROVED** [review]
cheez

**Review by willitcode — APPROVED** [review]

**Pull request merged**